src/threads/handlers.ts

```diff
@@ -0,0 +1,22 @@
+import type { APIGatewayProxyHandler } from 'aws-lambda';
+import { ForumTable } from './table';
+import { extractTokenData } from '../parsers';
+import { dynamodb } from '../dynamodb';
+
+const forumTable = new ForumTable(dynamodb);
+
+export const openThread: APIGatewayProxyHandler = async event => {
+  try {
+    const tokenData = extractTokenData(event);
+    const { participantId, itemId, userId } = tokenData;
+
+    await forumTable.addThreadEvent(participantId, itemId, {
+      type: 'thread_opened',
+      byUserId: userId,
+    });
+
+    return { statusCode: 201, body: '' };
+  } catch (error) {
+    return { statusCode: 401, body: '' }; // user is unauthenticated
```

Comment:
Any error (inc. JS, ...) will be interpreted as 401?

Reply:
The return type is actually not important since there is no response when sending a message to a socket service.
But yes.

Reply:
The return type is actually not important since there is no response when sending a message to a socket service.
But yes.